src/backend/Level0/CHIPBackendLevel0.hh

```diff
@@ -179,7 +178,7 @@ private:
   CHIPContextLevel0 *Ctx_;
   ze_event_pool_handle_t EventPool_;
   unsigned int Size_;
-  std::stack<std::shared_ptr<CHIPEventLevel0>> Events_;
+  std::stack<CHIPEventLevel0 *> Events_;
```

Comment:
Doesn't seem like the CHIPEventLevel0 objects are destructed along with LZEventPool. Relevant part from the destructor:
```c++
  while (Events_.size())
    Events_.pop();
```